### PR TITLE
LLM - Analytics prompt improvements

### PR DESCRIPTION
**Pull request by cgrellard-ledger**
### 📝 Description

Tracking the opt in prompt analytics (just the necessary user properties) when the user hasn't made a choice yet.

The `analyticsEnabled` and the `personalizedRecommendationsEnabled` state properties now have a default value of null instead of false to be able to differentiate users that already answered the Analytics Opt In Prompt. If they enable the analytics those values will be set to true, if they do not consent to analytics those values will be set to false.

### 📸 Demo
On the following video, we can see that we don't track events while the user didn't answer the analytics opt in prompt at the exception of the pages impressions and the clicks on this flow. In this case, we only send the necessary user properties needed by our SDKs (userId, optInAnalytics and optInPersonalRecommendations).
If the user consent to the analytics or personalized recommendations, we track every events with all other user properties.
If the user does not consent to the analytics or personalized recommendations, we do not track anything anymore. Not the app start and not the clicks on the analytics opt in prompt the next time it will be displayed to the user.
https://github.com/LedgerHQ/ledger-live/assets/17146928/d0e948b6-c85a-430d-8807-9f64f5900314

### ❓ Context

- **JIRA or GitHub link**: [LIVE-11202]

### ✅ Checklist

Pull Requests must pass the CI and be code reviewed. Set as Draft if the PR is not ready.

- [x] `npx changeset` was attached.
- [ ] **Covered by automatic tests.** 
- [ ] **Impact of the changes:** 
  - ...

---

### 🧐 Checklist for the PR Reviewers

- [ ] **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- [ ] **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- [ ] **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- [ ] **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- [ ] **Any new dependencies** have been justified and documented.
- [ ] **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

[LIVE-11202]: https://ledgerhq.atlassian.net/browse/LIVE-11202?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/ledger-live-mobile/src/screens/Settings/General/AnalyticsRow.tsx [hunk 1]
```diff
@@ -16,7 +16,7 @@ const AnalyticsRow = () => {
   const { t } = useTranslation();
   const dispatch = useDispatch();
   const [isOpened, setIsOpened] = useState(false);
-  const analyticsEnabled: boolean = useSelector(analyticsEnabledSelector);
+  const analyticsEnabled: boolean = !!useSelector(analyticsEnabledSelector);
```
Comment (mcayuelas-ledger) [inline apps/ledger-live-mobile/src/screens/Settings/General/AnalyticsRow.tsx:19]: Maybe it's better to use `Boolean()` instead 

https://www.tektutorialshub.com/typescript/typescript-boolean/#boolean-global-function

File: apps/ledger-live-mobile/src/hooks/useAnalyticsOptInPromptVariantB.ts [hunk 2]
```diff
@@ -23,10 +23,13 @@ const useAnalyticsOptInPrompt = () => {
...
     >();
-  const analyticsEnabled: boolean = useSelector(analyticsEnabledSelector);
-  const personalizedRecommendationsEnabled: boolean = useSelector(
+  const analyticsEnabled = useSelector(analyticsEnabledSelector);
+  const personalizedRecommendationsEnabled = useSelector(
     personalizedRecommendationsEnabledSelector,
   );
+  const isTrackingEnabled = analyticsEnabled || personalizedRecommendationsEnabled;
+  // When the user has not made a choice yet, we can track the analytics opt in flow
+  const shouldWeTrack = isTrackingEnabled === true || isTrackingEnabled === null;
```
Comment (mcayuelas-ledger) [inline apps/ledger-live-mobile/src/hooks/useAnalyticsOptInPromptVariantB.ts:32]:
```suggestion
const shouldWeTrack = !!isTrackingEnabled 
```

Comment (cgrellard-ledger) [reply apps/ledger-live-mobile/src/hooks/useAnalyticsOptInPromptVariantB.ts:32]: We need it to return true whether the isTrackingEnabled value is true or null, casting the value to a boolean won't work here

File: apps/ledger-live-mobile/src/reducers/settings.ts [hunk 3]
```diff
@@ -698,7 +698,7 @@ export const personalizedRecommendationsEnabledSelector = createSelector(
 );
 export const trackingEnabledSelector = createSelector(
   storeSelector,
-  s => s.analyticsEnabled || s.personalizedRecommendationsEnabled,
+  s => !!s.analyticsEnabled || !!s.personalizedRecommendationsEnabled,
```
[the remark this answers is not in the document]
Comment (cgrellard-ledger) [reply apps/ledger-live-mobile/src/reducers/settings.ts:701]: This one is used for a general purpose. In general, we do not track analytics if a user did not explicitly consent to the analytics (so if the `analyticsEnabled` and the `personalizedRecommendationsEnabled` are both null or false we should not track).
The comment above refers to the exception of the AnalyticsOptInPrompt where we are able to track analytics if the user did consent OR if he did not answered yet to the prompt.
For this exception above I am not using the `trackingEnabledSelector` but both the `analyticsEnabledSelector` and the `personalizedRecommendationsEnabledSelector`.

[the remark this answers is not in the document]
Comment (cgrellard-ledger) [reply apps/ledger-live-mobile/src/reducers/settings.ts:701]: Thanks to Kevin's feedback, the `analyticsEnabled` and the `personalizedRecommendationsEnabled` values cannot be null anymore. They are false by default and only become true when the user consent to them. In addition we now also have a `hasSeenAnalyticsOptInPrompt` property.

**Review by mcayuelas-ledger — DISMISSED** [review]

**Review by KVNLS — APPROVED** [review]

**Pull request merged**